code/modules/mob/living/say.dm

```diff
@@ -13,6 +13,11 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 	":c" = SQUAD_MARINE_3, ".c" = SQUAD_MARINE_3, "#c" = RADIO_CHANNEL_CLF_CMD,
 	":d" = SQUAD_MARINE_4, ".d" = SQUAD_MARINE_4, "#d" = RADIO_CHANNEL_CLF_CCT,
 	":e" = SQUAD_MARINE_5, ".e" = SQUAD_MARINE_5, "#e" = RADIO_CHANNEL_PMC_ENGI,
+	"#1" = SQUAD_UPP_1,
+	"#2" = SQUAD_UPP_2,
+	"#3" = SQUAD_UPP_3,
+	"#4" = SQUAD_UPP_4,
```

Comment:
please just add extra logic to handle using the existing a, b, c and d keys that routes based on faction